test/hotspot/jtreg/compiler/igvn/TestMinMaxIdeal.java

```diff
@@ -0,0 +1,141 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8373396
+ * @summary Verify that min/max add ideal optimizations get applied correctly
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib /
+ * @run driver ${test.main.class}
+ */
+
+package compiler.igvn;
+
+import compiler.lib.compile_framework.CompileFramework;
+import compiler.lib.template_framework.Template;
+import compiler.lib.template_framework.TemplateToken;
+import compiler.lib.template_framework.library.CodeGenerationDataNameType;
+import compiler.lib.template_framework.library.PrimitiveType;
+import compiler.lib.template_framework.library.TestFrameworkClass;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static compiler.lib.template_framework.Template.let;
+import static compiler.lib.template_framework.Template.scope;
+
+public class TestMinMaxIdeal {
+    public static void main(String[] args) {
+        // Create a new CompileFramework instance.
+        CompileFramework comp = new CompileFramework();
+
+        // Add a java source file.
+        comp.addJavaSourceCode("compiler.igvn.templated.MinMaxIdeal", generate(comp));
+
+        // Compile the source file.
+        comp.compile();
+
+        comp.invoke("compiler.igvn.templated.MinMaxIdeal", "main", new Object[] {new String[] {}});
+    }
+
+    private static String generate(CompileFramework comp) {
+        // Create a list to collect all tests.
+        List<TemplateToken> testTemplateTokens = Stream.of(Op.values())
+            .map(op -> new TestGenerator(op).generate())
+            .toList();
+
+        // Create the test class, which runs all testTemplateTokens.
+        return TestFrameworkClass.render(
+            // package and class name.
+            "compiler.igvn.templated", "MinMaxIdeal",
+            // List of imports.
+            Collections.emptySet(),
+            // classpath, so the Test VM has access to the compiled class files.
+            comp.getEscapedClassPathOfCompiledClasses(),
+            // The list of tests.
+            testTemplateTokens);
+    }
+
+    enum Op {
+        MIN_I("min", CodeGenerationDataNameType.ints()),
+        MAX_I("max", CodeGenerationDataNameType.ints()),
+        MIN_L("min", CodeGenerationDataNameType.longs()),
+        MAX_L("max", CodeGenerationDataNameType.longs());
+
+        final String functionName;
+        final PrimitiveType type;
+
+        Op(String functionName, PrimitiveType type) {
+            this.functionName = functionName;
+            this.type = type;
+        }
+    }
+
+    record TestGenerator(Op op) {
+        TemplateToken generate() {
+            var template = Template.make(() -> scope(
+                let("irNodeName", op.name()),
+                let("boxedTypeName", op.type.boxedTypeName()),
+                let("op", op.name()),
```

Comment:
Nit: you are repeating the same value `op.name()` with two hashtags `op` and `irNodeName`. Is that intentional?

Reply:
No, I'll get it fixed

Reply:
Just pushed a fix